eo-runtime/src/main/eo/org/eolang/positive-infinity.eo

```diff
@@ -59,70 +59,145 @@
         nan.as-bytes
 
   # Multiplication of $ and x
-  # @todo #1806:30min. Make positive-infinity.times and negative-infinity.times
-  #  worked with multiple arguments without using objects from eo-collections.
-  #  See positive-infinity.plus and negative-infinity.plus as an example
   [x...] > times
-    x.at 0 > first!
+    negative-infinity > neg-inf!
+    0 > zero!
 
     # here we use the rule "NaN != NaN is TRUE" to check if "num" is NaN
-    [num] > check-nan
+    [num] > is-nan
       not. > @
         num.eq num
 
+    [term] > is-nan-or-zero
+      or. > @
+        is-nan
+          term > t!
+        t.eq -0.0
+        t.eq 0.0
+        t.eq 0
+
+    [term] > is-term-gt-zero
+      try > @
+        []
+          term.gt 0 > @
+        [e]
+          term.gt 0.0 > @
+        FALSE
+
+    # @todo #1824:30min. Pass "is-nan-or-zero" to "times-rec" here and in
+    #  negative-infinity. Now for some reason it's not possible to pass two or
+    #  more objects-functions to another object. To avoid code duplication
```

Comment:
@maxonfjvipon As far as I understand this todo issue, you want to provide to the object more than 3 arguments. If it's it right you can just add all your arguments to new array and pass this array as argument to particular object